src/lib/recurringPurposes.js

```diff
@@ -64,6 +64,9 @@ export const findClosestStartAndEnd = (timeserie, contacts) => {
     minEndDistance = Number.MAX_SAFE_INTEGER
   for (const contact of contacts) {
     for (const address of contact.address) {
+      if (!address?.geo?.geo) {
+        continue
+      }
```

Comment:
If I'm not mistaken, we call this method from https://github.com/cozy/coachCO2/blob/43fa51de6d901d1478fae5c9bcda1a1c73cb4517/src/lib/recurringPurposes.js#L636C13-L636C13 

And here the var is called `contactsWithGeo` 

And this variable comes from https://github.com/cozy/coachCO2/blob/master/src/queries/queries.js#L303-L320 

How is it possible to have contacts without `geo.geo`?

Reply:
So, `contactsWithAtLeastOneAddressWithGeo`? ;)

Reply:
`IfYouLikeVarWithVeryExplicitNames`